doc/BorderRouter.rst

```diff
@@ -103,29 +107,30 @@ Packets currently identified for slow-path are:
 
 - SCMP traceroute packets
 
-Configuration
----------------
-
-The configuration of the border router will remain in the border router toml file.
-The following configuration entries are added:
-
 Pool size
 ^^^^^^^^^^^
 
-The size of the packet buffer pool of the receivers can be configured.
-An optimal value could be derived by calculating the maximum number of packets in-flight through the system:
+The pool size will be set by calculating the maximum number of packets in-flight through the system:
 
 .. code-block:: text
 
     pool_size := numReaders * readBatch + numProcessors * (processorQueueSize + 1) + numWriters * (writerQueueSize + writeBatchSize)
```

Comment:
I added it as a configuration option. But it would also be possible to just hardcode it.

Reply:
Matthias wrote:
"Recall that this is a userspace program and that we already have buffers on the underlying OS network stack both on the receive and transmit sides. We don't need to, and shouldn't, add more than minimal internal buffering.

The internal buffers should be big enough to accommodate for bursts caused by the internal structure, but not bigger. For the processing routines, the queue should be able to hold a read batch, and the sum of the queue sizes of the processing goroutines should be big enough to contain the read batches from all the reader routines.
The writer routines' queue should only be big enough for a single batch write.

**I would not expose this as a configuration option.** Operators should use the knob of the OS network stack first."

However, I don't really agree with this, it seems this buffering is way too small. E.g., two read processes might have both a batch of packets for the same processor. It seems to me that each queue for a processor should be able to handle batches from all readers. Also for the write routines, the proposed queue size seems too small.
What do you think?

Reply:
I think the queue of a processing routine should be able to hold more than just the burst size of a single read batch because otherwise in case of bursty traffic we might have to drop too many packets. Because of that I would suggest to just make the processing channel size configurable. (I already created a subsection in the design document)